File: ppt_gen/paths.py
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
THEMES_DIR = ROOT / "themes"
DECKS_DIR = ROOT / "decks"
BUILD_DIR = DECKS_DIR / ".build"
OUTPUT_DIR = ROOT / "output"
ASSETS_DIR = ROOT / "assets"
PLOTS_ASSETS_DIR = ASSETS_DIR / "plots"
DIAGRAMS_ASSETS_DIR = ASSETS_DIR / "diagrams"
CACHE_DIR = ASSETS_DIR / "cache"
DATA_DIR = ROOT / "data"
PLOTS_DIR = ROOT / "plots"
MERMAID_DIR = ROOT / "mermaid"


def theme_dir(name: str) -> Path:
    return THEMES_DIR / name


def ensure_dirs() -> None:
    for d in (BUILD_DIR, OUTPUT_DIR, PLOTS_ASSETS_DIR, DIAGRAMS_ASSETS_DIR, CACHE_DIR):
        d.mkdir(parents=True, exist_ok=True)

File: lib/paths.mjs
import { mkdir } from "node:fs/promises";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));

export const ROOT = join(__dirname, "..");
export const THEMES_DIR = join(ROOT, "themes");
export const DECKS_DIR = join(ROOT, "decks");
export const BUILD_DIR = join(DECKS_DIR, ".build");
export const OUTPUT_DIR = join(ROOT, "output");
export const ASSETS_DIR = join(ROOT, "assets");
export const PLOTS_ASSETS_DIR = join(ASSETS_DIR, "plots");
export const DIAGRAMS_ASSETS_DIR = join(ASSETS_DIR, "diagrams");
export const CACHE_DIR = join(ASSETS_DIR, "cache");
export const DATA_DIR = join(ROOT, "data");
export const MERMAID_DIR = join(ROOT, "mermaid");

export function themeDir(name) {
  return join(THEMES_DIR, name);
}

export async function ensureDirs() {
  for (const d of [
    BUILD_DIR,
    OUTPUT_DIR,
    PLOTS_ASSETS_DIR,
    DIAGRAMS_ASSETS_DIR,
    CACHE_DIR,
  ]) {
    await mkdir(d, { recursive: true });
  }
}
